### PR TITLE
[@awarns/phone-sensors]: Collection service injection

### PR DESCRIPTION
**Pull request by matey97**
This PR adds a new feature that allows working with different collection services. The included changes are the following:

- The `startDetectingPhoneSensorChangesTask(...)` and `stopDetectingPhoneSensorChangesTask(...)` generator functions use the collection service `BaseSensorRecordingService`.
- The `startDetectingPhoneNTPSyncedSensorChangesTask(...)` and `stopDetectingPhoneNTPSyncedSensorChangesTask(...)` generator functions use the collection service `NTPSyncedSensorRecordingService`. This service syncs the clock with an NTP server to accurately label (timestamp) the collected data.

-----
Hi @agonper, what do you think about having separated task generation functions? Another option is to add a parameter, but I think it would be less clear.

### CODE REVIEW

File: packages/phone-sensors/README.md [hunk 1]
```diff
@@ -17,15 +17,28 @@ ns plugin add @awarns/phone-sensors
 
 ## Usage
 
-After installing this plugin, you'll have access to two task groups to start and stop the data collection process.
+After installing this plugin, you'll have access to two task groups with two tasks each to start and stop the data collection process.
+The main difference between both tasks groups is the implementation of the underlying service that being used for the data collection.
```
Comment (agonper) [inline packages/phone-sensors/README.md:21]: I think something is missing in this sentence. "that ***is*** being used?"

**Review by agonper — COMMENTED** [review]
Looks good! Two things:
- I guess you decided to declare a new pair of task generators instead of using "just" an extra parameter in the existing one to mitigate the possibility of someone missing the extra parameter in the stop task, right? This way, they have to use the NTP or non-NTP tasks to start and stop.
- What's the NTP server for the NTP-based tasks? Can someone change it? I don't see how.

For the rest. Very good job, as always 😉

File: packages/phone-sensors/README.md (hunk 1 above)
Comment (matey97) [reply packages/phone-sensors/README.md:21]: Yep, that was missing

**Comment by matey97** [comment]
Hi @agonper, first of all, thanks for your review 😀. Regarding your comments:

> - I guess you decided to declare a new pair of task generators instead of using "just" an extra parameter in the existing one to mitigate the possibility of someone missing the extra parameter in the stop task, right? This way, they have to use the NTP or non-NTP tasks to start and stop.

Exactly. The option of injecting a parameter was a bit odd and more error-prone.

> - What's the NTP server for the NTP-based tasks? Can someone change it? I don't see how.

The NTP server is defined in the native dependency (`time.google.com`), and, as you saw, there is no way to change it. The reason for this was to avoid changing the server in one application and forgetting to change it in another one, thus having several synced applications but with different servers (different servers could provide different times 🤷‍♂️).

> For the rest. Very good job, as always 😉

Thank you!! 😁

**Comment by agonper** [comment]
Hi @matey97,

> The NTP server is defined in the native dependency (`time.google.com`), and, as you saw, there is no way to change it. The reason for this was to avoid changing the server in one application and forgetting to change it in another one, thus having several synced applications but with different servers (different servers could provide different times 🤷‍♂️).

Makes sense. I don't know if someone might want to use their time server at some point... Let's leave it as it is for now 🙂 

> Thank you!! 😁

You're welcome!

Do you want to merge this?

**Comment by matey97** [comment]
Hi @agonper,

> Do you want to merge this?

Yes, I think it's ready now

**Pull request merged**